### PR TITLE
Limelight subsystem

### PR DESCRIPTION
**Pull request by Jetblackdragon**
Added limelight subsystem with constructer and getter methods.

### CODE REVIEW

File: src/main/java/frc/team2412/robot/subsystems/LimelightSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	public static final double CAMERA_MOUNT_HEIGHT = 0.1143;
+	public static final double CAMERA_ANGLE_OFFSET = 0;
+	public static final double TARGET_HEIGHT = 0.33;
+
+	public static final double GOAL_DISTANCE_FROM_TARGET = 0.7;
+	public static final double GOAL_DISTANCE_FROM_NOTE = 0.3;
+
+	// MEMBERS
+
+	NetworkTable networkTable;
```
Comment (jamesdooley4) [inline src/main/java/frc/team2412/robot/subsystems/LimelightSubsystem.java]: Should be `final`

File: src/main/java/frc/team2412/robot/subsystems/LimelightSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+		Rotation2d targetHeading =
+				new Rotation2d(
+						currentPose.getRotation().getRadians() + Units.degreesToRadians(getHorizontalOffset()));
+		double targetDistance = getDistanceFromTargetTheSecond() / 39.3700787;
+
+		double targetX = Math.sin(targetHeading.getRadians()) * targetDistance;
+		double targetY = Math.cos(targetHeading.getRadians()) * targetDistance;
+
+		Pose2d targetPose =
+				new Pose2d(currentPose.getX() + targetY, currentPose.getY() + targetX, targetHeading);
```
Comment (jamesdooley4) [inline src/main/java/frc/team2412/robot/subsystems/LimelightSubsystem.java]: ? This is adding getX() to targetY, which seems odd. If that's intended, please add a comment explaining why.

**Review by jamesdooley4 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/team2412/robot/subsystems/LimelightSubsystem.java (hunk 1 above)
Comment (TAKBS2412) [reply src/main/java/frc/team2412/robot/subsystems/LimelightSubsystem.java]: (and `private`)

File: src/main/java/frc/team2412/robot/subsystems/LEDSubsystem.java
Comment (jamesdooley4) [inline src/main/java/frc/team2412/robot/subsystems/LEDSubsystem.java:1]: ? the LED subsystem is empty.

File: src/main/java/frc/team2412/robot/subsystems/LimelightSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import java.util.function.BiConsumer;
+
+/**
+ * All poses and transforms use the NWU (North-West-Up) coordinate system, where +X is
+ * north/forward, +Y is west/left, and +Z is up. On the field, this is based on the blue driver
+ * station (+X is forward from blue driver station, +Y is left, +Z is up).
+ *
+ * <p>At some point we should discuss how we want to handle the different alliances.
+ */
+public class VisionSubsystem extends SubsystemBase {
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/subsystems/LimelightSubsystem.java]: This should be LimelightSubsystem not VisionSubsystem?

File: src/main/java/frc/team2412/robot/subsystems/LimelightSubsystem.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+		var rotation =
+				new Rotation3d(Math.toRadians(pose[3]), Math.toRadians(pose[4]), Math.toRadians(pose[5]));
+		robotPose = new Pose3d(pose[0], pose[1], pose[2], rotation);
+		// TODO 3D odometry
+		poseConsumer.accept(robotPose.toPose2d(), time);
+	}
+
+	@Log
+	public boolean hasTargets() {
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/subsystems/LimelightSubsystem.java]: Assuming this is just for testing and there will be a method called (or field to return) instead?

**Review by bobtseattle — CHANGES_REQUESTED** [review]

**Comment by TAKBS2412** [comment]
Can we close this PR since no files have been changed in it?

**Pull request closed without merging**